src/otx/v2/adapters/torch/lightning/visual_prompt/engine.py

```diff
@@ -0,0 +1,284 @@
+"""OTX adapters.torch.lightning.anomalib.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from subprocess import run
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning.callbacks import (
+    EarlyStopping,
+    LearningRateMonitor,
+    ModelCheckpoint,
+    TQDMProgressBar,
+)
+from pytorch_lightning.core.datamodule import LightningDataModule
+from pytorch_lightning.loggers import CSVLogger
+from pytorch_lightning.loggers.logger import Logger
+from torch.utils.data import DataLoader
+
+from otx.v2.adapters.torch.lightning.engine import LightningEngine
+
+from .registry import VisualPromptRegistry
+
+if TYPE_CHECKING:
+    from pytorch_lightning.trainer.connectors.accelerator_connector import (
+        _PRECISION_INPUT,
+    )
+    from pytorch_lightning.utilities.types import EVAL_DATALOADERS
+
+PREDICT_FORMAT = Union[str, Path, np.ndarray]
+
+
+class VisualPromptEngine(LightningEngine):
+    """Anomalib engine using PyTorch and PyTorch Lightning."""
```

Comment:
Some docstrings need to be updated. Anomalib -> VisualPrompt

Reply:
Thank you. Fixed!